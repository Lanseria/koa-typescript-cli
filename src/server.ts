import * as Koa from "koa";
import * as Router from "koa-router";
import * as logger from "koa-logger";
import * as json from "koa-json";
import * as bodyParser from "koa-bodyparser";

const app = new Koa();
const router = new Router();

interface HelloRequest {
  name: String;
}

router.post("/", async (ctx, next) => {
  const data = <HelloRequest>ctx.request.body;
  ctx.body = { name: data.name };
  await next();
});

app.use(json());
app.use(logger());
app.use(bodyParser());

app.use(router.routes()).use(router.allowedMethods());

app.listen(3000, () => {
  console.log("Server running on port http://localhost:3000");
});
