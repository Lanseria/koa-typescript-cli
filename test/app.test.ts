import request from "supertest";
import { eApp } from "../src/server";
import Result from "../src/vo/result.vo";

test("Hello koa-typescript-cli", async () => {
  const app = await eApp;
  const response: request.Response = await request(app.callback()).get("/");
  expect(response.status).toBe(200);
  const result: Result<string> = JSON.parse(response.text);
  expect(result.data).toBe("hello world koa-typescript-cli");

  expect(response.text).toMatchSnapshot();
});
