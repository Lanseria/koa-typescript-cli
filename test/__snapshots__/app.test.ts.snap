// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello koa-typescript-cli 1`] = `"{\\"code\\":0,\\"msg\\":\\"success\\",\\"data\\":\\"hello world koa-typescript-cli\\"}"`;

exports[`Hello world works 1`] = `"{\\"code\\":0,\\"msg\\":\\"success\\",\\"data\\":[{\\"id\\":\\"51127ff5-11de-4797-829e-17ad5e02a510\\",\\"name\\":\\"heelo\\",\\"releaseYear\\":null,\\"rating\\":null}]}"`;
